### PR TITLE
1/ install.mdx -> install.md + note component

### PR DESCRIPTION
**Pull request by yuhan**
## Summary & Motivation
this pr migrates our very first page from mdx to markdoc.

as we go through the content migrate, we also migrate components from mdx components to markdoc components as we encounter them. here's the component old vs new:
| MDX (old) | Markdoc (new) |
| ------------- | ------------- |
| Partial `DagsterVersion` which has to be a UI component | `{% partial file="DagsterVersion.md" /%}` which can reference a md file  |
| `<Note><strong>Note</strong>: blah</Note>` which doesn't respect the inline markdown  | `{% note %}**Note**: blah{% /note %}` which respect the inner markdown format |
| Retiring `ArticleList` and `ArticleListItem` | `{% table %}` (markdoc built-in component [doc](https://markdoc.dev/docs/tags#table-without-headings)) this also caught a long-standing broken link which has been missed in the old ArticleList component - see [2]|

This PR also maintains backcompat with MDX components in [1], and implements the test for the new markdown files.

## How I Tested These Changes
- link renders the same as before: https://01-09-1--install-mdx---install-md-note-component.dagster.dagster-docs.io/getting-started/install
- unit tests passes

### CODE REVIEW

**Comment by yuhan** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19094?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #19093** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19093?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #19094** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19094?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19094?utm_source=stack-comment).

File: docs/next/components/mdx/MDXComponents.tsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ import Zoom from 'react-medium-image-zoom';
 
 import Icons from '../Icons';
 import Link from '../Link';
+import {Note, Warning} from '../markdoc/Callouts';
```
Comment (yuhan) [inline docs/next/components/mdx/MDXComponents.tsx:19]: [1]

File: docs/next/pages/getting-started/install.md [hunk 2]
```diff
@@ -67,21 +65,16 @@ poetry add dagster dagster-webserver
...
-    title="Contributing to Dagster"
-    href="/community/contributing"
-  ></ArticleListItem>
-</ArticleList>
+{% table %}
+
+---
+
+- [Creating a new Dagster project](/getting-started/create-new-project)
+- [Dagster project files](/guides/understanding-dagster-project-files)
```
Comment (yuhan) [inline docs/next/pages/getting-started/install.md:73]: [2]

File: docs/next/__tests__/mdxInternalLinks.test.ts [hunk 3]
```diff
@@ -49,29 +45,42 @@ test('No dead navs', async () => {
...
     }),
   );
+  // Parse md files to find all internal links and populate the store
+  await Promise.all(
+    allMdFilePaths.map(async (relativeFilePath) => {
+      const absolutePath = path.resolve(DOCS_DIR, relativeFilePath);
+      const fileContent = await fs.promises.readFile(absolutePath, 'utf-8');
+      // separate content and front matter data
+      const {content} = matter(fileContent);
+      // console.log(relativeFilePath, remark().use(mdx).parse(content));
```
Comment (hellendag) [inline docs/next/__tests__/mdxInternalLinks.test.ts:72]: Delete?

**Review by hellendag — COMMENTED** [review]
Looks fine to me, will let @erinkcochran87 take a look too :)

File: docs/next/pages/getting-started/install.md [hunk 4]
```diff
@@ -22,11 +20,11 @@ pip --version
 
 ## Installing Dagster into an existing Python environment
 
-<Note>
-  <strong>Note</strong>: We strongly recommend installing Dagster inside a
-  Python virtualenv. If running Anaconda, install Dagster inside a Conda
-  environment.
-</Note>
+{% note %}
```
Comment (erinkcochran87) [inline docs/next/pages/getting-started/install.md:23]: ❤️

Comment (erinkcochran87) [reply docs/next/pages/getting-started/install.md:23]: I'm having some trouble testing this locally (probably an env issue with the markdoc stuff), but d'you know if the new notes/callouts will preserve line breaks between paragraphs? Ex:

```
{% note %}
One line!

Another line!

Woo!
{% /note %}
```

This looks fine when rendered even though the migrated copy is on three lines (why is that, btw?), but just curious about multiple paragraphs being preserved.

**Review by erinkcochran87 — APPROVED** [review]
This is so great <3 I left a few questions/comments, but they're not showstoppers for this PR - we can merge and tweak them later.

File: docs/next/pages/getting-started/install.md (hunk 4 above)
Comment (yuhan) [reply docs/next/pages/getting-started/install.md:23]:
```
{% note %}

One line!

Another line!

Woo!

{% /note %}
```
would render as this:
image

---

the migrated copy is on three lines but without empty new lines between each line, so it renders as one line. it's exactly how it was before which i guess was just a legacy formatting result.

Comment (erinkcochran87) [reply docs/next/pages/getting-started/install.md:23]: excellent, ty!

File: docs/next/pages/getting-started/install.md [hunk 5]
```diff
@@ -67,21 +65,16 @@ poetry add dagster dagster-webserver
...
-  ></ArticleListItem>
-</ArticleList>
+{% table %}
+
+---
+
+- [Creating a new Dagster project](/getting-started/create-new-project)
+- [Dagster project files](/guides/understanding-dagster-project-files)
+
+---
```
[the remark this answers is not in the document]
Comment (yuhan) [reply docs/next/pages/getting-started/install.md:75]: per offline discussion, will do a follow up to resolve this one.

**Comment by yuhan** [comment]
### Merge activity

* **Jan 18, 2:43 PM**: @yuhan started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19094).

**Pull request merged**